### PR TITLE
Feature/implement filings

### LINKED ISSUE #6: Implement Filings
Ability to list all filings with appropriate descriptions from:

https://github.com/companieshouse/api-enumerations/blob/master/filing_history_descriptions.yml

### PR DESCRIPTION
**Pull request by turowicz**
Resolves #6 

Code ready
Unit tests ready
Integration tests due tomorrow

Please have a look at what I've done. I tried to follow all your patterns as much as I could.

### CODE REVIEW

**Comment by turowicz** [comment]
@kevbite please review the PR and help with failing build. I will add integration tests tomorrow.
cc @rmourato @Jay2Base

**Comment by kevbite** [comment]
Looks like its failing due to a conflict in versions, If you update the appveyor.yml file to version 1.3.0 it should start playing nicely.


```yml
-
  branches:
    only:
      - master
   
  version: 1.3.0
  
  install:
```

**Comment by kevbite** [comment]
PR looks totally fine, once we've got an integration test around it we'll merge it in to master. Thanks for taking time to contribute.

**Comment by turowicz** [comment]
@kevbite integration tests added, please review.

File: src/LiberisLabs.CompaniesHouse.IntegrationTests/Tests/CompanyFilingHistoryTests/CompanyFilingHistoryTestsInvalid.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        [SetUp]
+        protected override void When()
+        {
+            WhenRetrievingAnInvalidCompanyProfile();
+        }
+
+        [Test]
+        public void ThenTheProfileIsNotReturned()
+        {
+            Assert.That(_result.Data.Items, Is.Null);
```
Comment (kevbite) [inline src/LiberisLabs.CompaniesHouse.IntegrationTests/Tests/CompanyFilingHistoryTests/CompanyFilingHistoryTestsInvalid.cs]: Shouldn't `_result.Data` be null? as having the `_result.Data.Items` null makes that a bit ambiguous as whats the difference between a empty list and a null list?

if we could change this to just pass up the `_result.Data` back as null that be great! keeps it consistent with the Company Profile client too - https://github.com/LiberisLabs/CompaniesHouse.NET/blob/master/src/LiberisLabs.CompaniesHouse.IntegrationTests/Tests/CompanyProfileTests/CompanyProfileTestsInvalid.cs#L19

Comment (turowicz) [inline src/LiberisLabs.CompaniesHouse.IntegrationTests/Tests/CompanyFilingHistoryTests/CompanyFilingHistoryTestsInvalid.cs]: Hey - sure I can, but the actual data is not null. It is an envelope response with row counts, page number etc. If the company doesn't exist there is even a new status `filing-history-not-available-invalid-format` which could be used by the client apps.

Comment (turowicz) [inline src/LiberisLabs.CompaniesHouse.IntegrationTests/Tests/CompanyFilingHistoryTests/CompanyFilingHistoryTestsInvalid.cs]: Let me know what you think with that information @kevbite. I can make it null if needed.

Comment (kevbite) [inline src/LiberisLabs.CompaniesHouse.IntegrationTests/Tests/CompanyFilingHistoryTests/CompanyFilingHistoryTestsInvalid.cs]: Ah right i see, I thought we was explicitly setting it to null within `CompaniesHouseCompanyFilingHistoryClient` when it was a 404 but this is actually the response from the web service? Lets leave it how it is and I'll merge it in.

Comment (turowicz) [inline src/LiberisLabs.CompaniesHouse.IntegrationTests/Tests/CompanyFilingHistoryTests/CompanyFilingHistoryTestsInvalid.cs]: Yes - there is always a response. For a valid company it has an array of items, for invalid it has a null array, but it always have the encapsulating object with meta information about results.

**Comment by turowicz** [comment]
Thanks @kevbite !

**Comment by kevbite** [comment]
All published to nuget now too - https://www.nuget.org/packages/CompaniesHouse/1.3.0

**Pull request merged**